### PR TITLE
Add remark for JsonNumberHandling of Special Numbers in System.Text.Json

### PR DESCRIPTION
**Pull request by deeprobin**
## Summary

See dotnet/runtime#48926

/cc @layomia
/cc @GrabYourPitchforks

### CODE REVIEW

File: xml/System.Text.Json.Serialization/JsonNumberHandling.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
   </Attributes>
   <Docs>
     <summary>Determines how <see cref="T:System.Text.Json.JsonSerializer" /> handles numbers when serializing and deserializing.</summary>
-    <remarks>To be added.</remarks>
+    <remarks>You can do this, but you need to be aware that this will create potentially non-interoperable JSON.</remarks>
```
Comment (krwq) [inline xml/System.Text.Json.Serialization/JsonNumberHandling.xml:24]: https://github.com/dotnet/runtime/pull/62690#pullrequestreview-831384977

File: xml/System.Text.Json.Serialization/JsonNumberHandling.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
   </Attributes>
   <Docs>
     <summary>Determines how <see cref="T:System.Text.Json.JsonSerializer" /> handles numbers when serializing and deserializing.</summary>
-    <remarks>To be added.</remarks>
+    <remarks>Mentioned values are not defined by the specification. Using this setting may potentially produce JSON which may produce JSON which cannot be parsed by other JSON implementations.</remarks>
```
Comment (gewarren) [inline xml/System.Text.Json.Serialization/JsonNumberHandling.xml:24]:
```suggestion
    <remarks>The fields of this enumeration are not defined by the JSON specification. Altering the default number handling can potentially produce JSON that cannot be parsed by other JSON implementations.</remarks>
```

File: xml/System.Text.Json.Serialization/JsonNumberHandling.xml [hunk 3]
```diff
@@ -109,7 +109,7 @@
       <MemberValue>2</MemberValue>
       <Docs>
         <summary>Numbers will be written as JSON strings (with quotes), not as JSON numbers.</summary>
-        <remarks>To be added.</remarks>
+        <remarks>Mentioned values are not defined by the specification. Using this setting may potentially produce JSON which may produce JSON which cannot be parsed by other JSON implementations.</remarks>
```
Comment (gewarren) [inline xml/System.Text.Json.Serialization/JsonNumberHandling.xml:112]: Why is this remark here and not in the other fields too? But I think it's fine just to have it in the main remarks for the enumeration.

Comment (deeprobin) [reply xml/System.Text.Json.Serialization/JsonNumberHandling.xml:112]: The other fields are not included in spec. 
See https://github.com/dotnet/runtime/issues/48926#issuecomment-973302102

@layomia Maybe you can say something about it :D

File: xml/System.Text.Json.Serialization/JsonNumberHandling.xml [hunk 4]
```diff
@@ -21,7 +21,7 @@
   </Attributes>
   <Docs>
     <summary>Determines how <see cref="T:System.Text.Json.JsonSerializer" /> handles numbers when serializing and deserializing.</summary>
-    <remarks>To be added.</remarks>
+    <remarks>The fields of this enumeration are not defined by the JSON specification. Altering the default number handling can potentially produce JSON that cannot be parsed by other JSON implementations.</remarks>
```
Comment (gewarren) [inline xml/System.Text.Json.Serialization/JsonNumberHandling.xml:24]: @deeprobin Is this remark meant to apply to `AllowNamedFloatingPointLiterals`? If so, it should be moved to line 46. Also, can you change it to say "This field is not defined by...".

**Comment by gewarren** [comment]
One thing to note is that currently, remarks on enum fields aren't displayed in the docs anywhere. You could add a general remark to the enum as a whole, and just call out the two fields specifically.

**Comment by deeprobin** [comment]
> One thing to note is that currently, remarks on enum fields aren't displayed in the docs anywhere. You could add a general remark to the enum as a whole, and just call out the two fields specifically.

@layomia What is your approach to this?

**Review by gewarren — APPROVED** [review]

**Comment by deeprobin** [comment]
> > One thing to note is that currently, remarks on enum fields aren't displayed in the docs anywhere. You could add a general remark to the enum as a whole, and just call out the two fields specifically.
> 
> @layomia What is your approach to this?

@gewarren See https://github.com/dotnet/runtime/pull/62690#discussion_r774182347

**Review by gewarren — APPROVED** [review]

**Comment by gewarren** [comment]
Thanks @deeprobin

**Pull request merged**